osu.Game/Users/UserPanel.cs

```diff
@@ -68,7 +70,8 @@ private void load()
             base.Action = ViewProfile = () =>
             {
                 Action?.Invoke();
-                profileOverlay?.ShowUser(User);
+                if (ShowProfileOnClick)
+                    profileOverlay?.ShowUser(User);
```

Comment:
Not a fan of this to be fair. I'd prefer this to be something like

```csharp
            base.Action = () =>
            {
                if (Action != null)
                    Action.Invoke();
                else
                    profileOverlay?.ShowUser(User);
            };
```

with xmldoc saying that specifying no `Action` defaults to opening profile overlay. Or heck, even initialise `Action` to show overlay by default and just call that.

Reply:
I did it this way because the other usages of this are compounding a new action in addition to showing the profile overlay (ie. the login dropdown hiding itself on display). Also thinking that we don't really want to block the "show profile" behaviour under normal scenarios. Maybe need to re-think this display and add a spectate button below or next to the card instead.